bittensor/utils/__init__.py

```diff
@@ -102,7 +105,126 @@ def solve_for_difficulty( block_hash, difficulty ):
             break
     return nonce, seal
 
-def solve_for_difficulty_fast( subtensor, wallet, num_processes: int = None, update_interval: int = 500000 ) -> Tuple[int, int, Any, int, Any]:
+
+def get_human_readable(num, suffix="H"):
+    for unit in ["", "K", "M", "G", "T", "P", "E", "Z"]:
+        if abs(num) < 1000.0:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1000.0
+    return f"{num:.1f}Y{suffix}"
+
+
+def millify(n: int):
+    millnames = ['',' K',' M',' B',' T']
+    n = float(n)
+    millidx = max(0,min(len(millnames)-1,
+                        int(math.floor(0 if n == 0 else math.log10(abs(n))/3))))
+
+    return '{:.0f}{}'.format(n / 10**(3 * millidx), millnames[millidx])
+
+@dataclass
+class POWSolution:
+    """A solution to the registration PoW problem."""
+    nonce: int
+    block_number: int
+    difficulty: int
+    seal: bytes
+
+class Solver(multiprocessing.Process):
+    proc_num: int
+    num_proc: int
+    update_interval: int
+    best_queue: multiprocessing.Queue
+    time_queue: multiprocessing.Queue
+    solution_queue: multiprocessing.Queue
+    newBlockEvent: multiprocessing.Event
+    stopEvent: multiprocessing.Event
+    curr_block: multiprocessing.Array
+    curr_block_num: multiprocessing.Value
+    curr_diff: multiprocessing.Array
+    check_block: multiprocessing.Lock
+    limit: int
+
```

Comment:
Would love some documentation on this. This inherits from a multiprocessing process, correct? Would love some explanation of the different queues.